pay-api/src/pay_api/services/eft_service.py

```diff
@@ -59,15 +61,11 @@ def apply_credit(self,
                                       payment_date=payment_date,
                                       paid_amount=invoice_balance - new_invoice_balance)
 
-        invoice_ref = self.create_invoice_reference(invoice=invoice_model, payment=payment)
-        receipt = self.create_receipt(invoice=invoice_model, payment=payment)
+        invoice_reference = self.create_invoice_reference(invoice=invoice_model, payment=payment)
```

Comment:
The invoice reference is now being created when we create an invoice, we don't need to create it again when applying the credit.